src/SixLabors.Fonts/StreamFontMetrics.TrueType.cs

```diff
@@ -21,11 +21,31 @@ namespace SixLabors.Fonts
     /// </content>
     internal partial class StreamFontMetrics
     {
+        /// <summary>
+        /// <para>
+        /// TODO: See if something can be done about this. For potential workarounds <see href="http://rastertragedy.com/RTRCh4.htm#Sec1"/>
+        /// These common fonts can be hinted by the engine only in XY mode and only if we do not scale the x-dimension.
+        /// even then the output is worse than the non-hinted version due to thinning of certain strokes.
+        /// <see href="http://agg.sourceforge.net/antigrain.com/research/font_rasterization/"/>.
+        /// </para>
+        /// <para>
+        /// We'll sacrifice hinting for these fonts to allow the engine to work better for modern fonts and avoid potentially raised issues.
+        /// </para>
+        /// </summary>
+        private static readonly string[] BadHintList = { "Arial", "Times New Roman", "Tahoma", "Palatino Linotype", "Segoe UI" };
+        private bool? shouldHint;
```

Comment:
There's a LOT of knowledge in this document, unfortunately I couldn't find a way to get these fonts to hint well. They're all older ones that have known issues with hinting so to save effort I've disabled hinting of them for now.

http://rastertragedy.com/RTRCh4.htm#Sec1

Every single modern font I tested seems to work really well.

Reply:
is there any metadata in the font we can use to determine if we should enable hinting? some created date maybe or so versioning flag difference? anything to save just hard coding a set of font names.

Reply:
Unfortunately not. 😔

There's some guidance for how to actually mitigate the issues I saw but I spent a few days and couldn't translate it into code. Nor could I find reference in FreeType that provided a complete workaround matching that guidance.

> Accordingly I was looking for an approach that combines the advantages of overscaling before constraint application with the “safety” of overscaling after constraint application. To do so I “jury rigged” the rasterizer to dynamically re-interpret all rounding instructions to mean “round to sample boundary” if the rendering method is ClearType and if the direction of the projection vector p is not parallel to the long side of the LCD sub-pixels, else to keep the meaning “round to pixel boundary” as before:
> 
> - Apply contextually interpreted constraints (“hints”) to outlines
> - Overscale constrained outlines in x-direction by 6x
> - Sample overscaled constrained outlines into overscaled bitmaps
> - Apply anti-aliasing filter to downsample overscaled bitmaps
> 
> Semantically, the two interpretations of rounding instructions are most similar. In fact, if you consider full-pixel bi-level rendering (colloquially “black-and-white” rendering) as “oversampling” at the rate of one sample per pixel, then the two interpretations are identical. Practically, this approach eliminates the “explosions” and it overcomes the absence of a TrueType instruction that would permit “finer grained” rounding than to the nearest 1/2 of a pixel (cf [4.3.0](http://rastertragedy.com/RTRCh4.htm#Sec30)).

Reply:
I do wonder whether we just just try tagging the chap who built the original interpreter to see if he knows how?

Reply:
I hate to do this and apologies in advance but @MikePopoloski would you have any insight into how to implement some of these workarounds?

Reply:
Oh hello. No sorry, I haven't thought much about font rendering in like 7 years. Glad someone is making use of the interpreter though, it's a neat bit of code and there aren't (or at least weren't when I built it) too many open source implementations of a TT interpreter.

Reply:
Thanks for replying! That was a bit of a hit and hope tagging you like that. 

I think there's very few people in the world who actually understand the TT interpreter. I've yet to find an implementation other than yours in the .NET ecosystem and elsewhere there just appears to be using Freetype.

I'll persevere though and continue to do my own research. Fongers crossed I can figure it all out. It's an interesting challenge!